lib/queue.js

```diff
@@ -89,7 +94,7 @@ class PeerQueue extends EventEmitter {
 
     // TODO: support a multiplex: true flag, that will make the info object emit a
     // 'topic' event instead of making dup connections, per topic
-    const id = toID(peer) + (peer.topic ? '@' + peer.topic.toString('hex') : '')
+    const id = toID(peer)
```

Comment:
ohh, wow are we saying peers weren't being removed because ID generation was different here vs `remove` method

Reply:
This is only a part of the problem. Other part is that `remove` was not called at all :)

Reply:
At least not in case of reconnects or bans.

Reply:
yeah - nice

Reply:
that was my bad. nice fix.